pkg/controllermanager/controller/backupinfrastructure/backup_infrastructure_control.go

```diff
@@ -109,13 +109,17 @@ func (c *Controller) reconcileBackupInfrastructureKey(key string) error {
 	if reconcileErr != nil {
 		durationToNextSync = 15 * time.Second
 	} else if deleted {
-		gracePeriod := computeGracePeriod(c.config)
-		durationToActualDeletion := gracePeriod - time.Now().Sub(backupInfrastructure.DeletionTimestamp.Time)
-		// We don't set durationToNextSync directly to durationToActualDeletion since,
-		// we want reconciliation to update status as per sync period. This will help in adjusting
-		// the next sync time in case deletionGracePeriod is reduced from GCM config.
-		if durationToNextSync > durationToActualDeletion {
-			durationToNextSync = durationToActualDeletion
+		if kutil.HasMetaDataAnnotation(&backupInfrastructure.ObjectMeta, common.BackupInfrastructureForceDelete, "true") {
```

Comment:
Why do we have that? That would mean that if I create a `BackupInfrastructure` resource with this annotation it would get reconciled every `15s`, creating a lot of API calls.
Why do we need that at all? Can't we skip it because we are anyway increasing the generation if the annotation got set? Also, if an error occurs during deletion we requeue in `15s` already.

Reply:
Nope, it's under `if deleted` case. So, codepath will reach this step only when deletionTimestamp as well as annotation is set which will occur only once. By the time it reaches here the backupInfra finaliser will be surely removed. So, in next reconciliation we won't find resource. And it won't get added to queue again.

Reply:
OK, but still: Why is this needed? Why do we need this automatic/frequent requeueing?
When the annotation gets set and the generation increases then the controller will reconcile anyway - independent of whether the deletion was pending/scheduled for some other time?!

Reply:
Ok. Will remove it.

Reply:
Do you agree? Am I right with this assumption?

Reply:
Agreed.